rpcapi/votehandlers.go

```diff
@@ -25,11 +25,11 @@ func (r *RPCAPI) submitRawTx(request *api.APIrequest) (*api.APIresponse, error)
 		return nil, fmt.Errorf("no reply from vochain")
 	}
 	if res.Code != 0 {
-		return nil, fmt.Errorf("%s", res.Data)
+		return nil, fmt.Errorf("%s", string(res.Data))
 	}
 	log.Debugf("broadcasting tx hash:%s", res.Hash)
 	// return nullifier or other info
-	return &api.APIresponse{Payload: fmt.Sprintf("%x", res.Data)}, nil
+	return &api.APIresponse{Payload: fmt.Sprintf("%s", string(res.Data))}, nil
```

Comment:
This change is invalid; the non-error response will be binary and thus needs to be encoded as hex, so you can't simply return the non-encoded form as a JSON string. You can see how the tests broke because of this.

I'd say just revert this line change.